v3/src/components/case-table/index-menu-list.tsx

```diff
@@ -1,11 +1,13 @@
 import { MenuItem, MenuList, useToast } from "@chakra-ui/react"
 import React from "react"
+import { IDataSet } from "../../data-model/data-set"
 
 interface IProps {
   caseId: string
   index?: number
+  data?: IDataSet
```

Comment:
Rather than passing in the `data` as a prop, you should use the `useDataSetContext` hook.